cmd/admin-revoker/main.go

```diff
@@ -92,578 +26,49 @@ type Config struct {
-	var h []string
-	_, err := r.dbMap.Select(ctx, &h, "SELECT certSerial FROM keyHashToSerial WHERE keyHash = ?", spkiHash[:])
-	if err != nil {
-		if db.IsNoRows(err) {
-			return nil, berrors.NotFoundError("no certificates with a matching SPKI hash were found")
-		}
-		return nil, err
-	}
-	return h, nil
-}
-
-// This abstraction is needed so that we can use sort.Sort below
-type revocationCodes []revocation.Reason
-
-func (rc revocationCodes) Len() int           { return len(rc) }
-func (rc revocationCodes) Less(i, j int) bool { return rc[i] < rc[j] }
-func (rc revocationCodes) Swap(i, j int)      { rc[i], rc[j] = rc[j], rc[i] }
-
-func privateKeyBlock(ctx context.Context, r *revoker, dryRun bool, comment string, count int, spkiHash core.Sha256Digest, keyPath string) error {
-	keyExists, err := r.spkiHashInBlockedKeys(ctx, spkiHash)
-	if err != nil {
-		return fmt.Errorf("while checking if the provided key already exists in the 'blockedKeys' table: %s", err)
-	}
-	if keyExists {
-		return errors.New("the provided key already exists in the 'blockedKeys' table")
-	}
-
-	if dryRun {
-		r.log.AuditInfof(
-			"To block issuance for this key and revoke %d certificates via bad-key-revoker, run with -dry-run=false",
-			count,
-		)
-		r.log.AuditInfo("No keys were blocked or certificates revoked, exiting...")
-		return nil
-	}
-
-	r.log.AuditInfo("Attempting to block issuance for the provided key")
-	err = r.blockByPrivateKey(context.Background(), comment, keyPath)
-	if err != nil {
-		return fmt.Errorf("while attempting to block issuance for the provided key: %s", err)
-	}
-	r.log.AuditInfo("Issuance for the provided key has been successfully blocked, exiting...")
-	return nil
-}
-
-func privateKeyRevoke(r *revoker, dryRun bool, comment string, count int, keyPath string) error {
-	if dryRun {
-		r.log.AuditInfof(
-			"To immediately revoke %d certificates and block issuance for this key, run with -dry-run=false",
-			count,
-		)
-		r.log.AuditInfo("No keys were blocked or certificates revoked, exiting...")
-		return nil
-	}
-
-	if count <= 0 {
-		// Do not revoke.
-		return nil
-	}
-
-	// Revoke certificates.
-	r.log.AuditInfof("Attempting to revoke %d certificates", count)
-	err := r.revokeByPrivateKey(context.Background(), keyPath)
-	if err != nil {
-		return fmt.Errorf("while attempting to revoke certificates for the provided key: %s", err)
-	}
-	r.log.AuditInfo("All certificates matching using the provided key have been successfully")
-
-	// Block future issuance.
-	r.log.AuditInfo("Attempting to block issuance for the provided key")
-	err = r.blockByPrivateKey(context.Background(), comment, keyPath)
-	if err != nil {
-		return fmt.Errorf("while attempting to block issuance for the provided key: %s", err)
-	}
-	r.log.AuditInfo("All certificates have been successfully revoked and issuance blocked, exiting...")
-	return nil
-}
-
 func main() {
-	usage := func() {
-		fmt.Fprint(os.Stderr, usageString)
-		os.Exit(1)
-	}
-	if len(os.Args) <= 2 {
```

Comment:
By removing lines 529 - 535, `admin-revoker` will panic when invoked as `./bin/admin-revoker`. I think it would be better if the `usage` func said to use `admin` instead.
```
root@3ed4c0f24b0a:/boulder# ./bin/admin-revoker 
2024-02-07T20:41:40.008345+00:00Z 3ed4c0f24b0a unknown admin-revoker[3092]: 3 admin-revoker 97Du-Qs [AUDIT] Panic caused by err: runtime error: index out of range [1] with length 1
2024-02-07T20:41:40.008411+00:00Z 3ed4c0f24b0a unknown admin-revoker[3092]: 3 admin-revoker hJvspg4 [AUDIT] Stack Trace (Current goroutine) goroutine 1 [running]:\nruntime/debug.Stack()\n	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e\ngithub.com/letsencrypt/boulder/cmd.AuditPanic()\n	/boulder/cmd/shell.go:379 +0xe8\npanic({0x14e6ec0?, 0xc00013f7d0?})\n	/usr/local/go/src/runtime/panic.go:914 +0x21f\ngithub.com/letsencrypt/boulder/cmd/admin-revoker.main()\n	/boulder/cmd/admin-revoker/main.go:30 +0x6c5\nmain.main()\n	/boulder/cmd/boulder/main.go:133 +0x3fc\n
```